Client/Application/SceneDelegate.swift

```diff
@@ -41,8 +41,15 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
         guard !AppConstants.isRunningUnitTest else { return }
 
         if AppConstants.useCoordinators {
-            sceneCoordinator = SceneCoordinator()
-            sceneCoordinator?.start(with: scene)
+            self.window = configureWindowFor(scene)
+            let navigationController = createNavigationController()
+            let router = DefaultRouter(navigationController: navigationController)
+
+            sceneCoordinator = SceneCoordinator(router: router)
+            sceneCoordinator?.start()
+
+            window?.rootViewController = navigationController
+            window?.makeKeyAndVisible()
```

Comment:
It feels a little weird to me that the creation of the navigation controller and the setting of the root view controller are handled out here, I think they should probably be the responsibility of the coordinator. Maybe the scene coordinator can take it's window as an init property instead of the navigation controller?
I don't feel strongly about this tho so if you disagree feel free to keep it as is.

Reply:
Yeah that's what I had tried in a way, passing the scene into the scene coordinator to setup the navigation controller, but then I had to put the `router` as optional on the `Coordinator` protocol since this `SceneCoordinator` couldn't be init with its router right away. Having the `router` optional makes things harder down the road since we need to force it to pass to other coordinators with `!`. Maybe the SceneCoordinator shouldn't be following Coordinator protocol since it wouldn't have a router on the init, but then it feels weird that way too. Let me know if you see ways of solving this.

Reply:
Can the router not be initialised inside the coordinators init?

Reply:
Ok I made some changes, basically my issue was that I couldn't call self to call the helper methods to `createNavigationController` and `configureWindowFor`. So I added a SceneSetupHelper for this 🤔. I think this works, but let me know if it's not good enough